larpandoracontent/LArMonitoring/HierarchyValidationAlgorithm.cc

```diff
@@ -0,0 +1,400 @@
+        const int nUnmatched{static_cast<int>(matchInfo.GetUnmatchedMC().size())};
+        const int nNodes{static_cast<int>(matchInfo.GetNMCNodes())};
+        int hasLeadingLepton{0}, isLeadingLeptonCorrect{0};
+
+        std::set<const LArHierarchyHelper::MCHierarchy::Node *> trackNodeSet, showerNodeSet;
+        int nGoodTrackMatches{0}, nGoodShowerMatches{0};
+        for (const LArHierarchyHelper::MCMatches &mcMatch : matchInfo.GetGoodMatches())
+        {
+            const LArHierarchyHelper::MCHierarchy::Node *pNode{mcMatch.GetMC()};
+            if (pNode->IsLeadingLepton())
+            {
+                hasLeadingLepton = 1;
+                isLeadingLeptonCorrect = 1;
+            }
+            const int pdg{std::abs(pNode->GetParticleId())};
+            if (pdg == PHOTON || pdg == E_MINUS)
+            {
+                showerNodeSet.insert(pNode);
+                ++nGoodShowerMatches;
+            }
+            else
+            {
+                trackNodeSet.insert(pNode);
+                ++nGoodTrackMatches;
+            }
+        }
+
+        int nAboveThresholdTrackMatches{0}, nAboveThresholdShowerMatches{0};
+        for (const LArHierarchyHelper::MCMatches &mcMatch : matchInfo.GetAboveThresholdMatches())
+        {
+            const LArHierarchyHelper::MCHierarchy::Node *pNode{mcMatch.GetMC()};
+            if (pNode->IsLeadingLepton())
+                hasLeadingLepton = 1;
+            const int pdg{std::abs(pNode->GetParticleId())};
+            if (pdg == PHOTON || pdg == E_MINUS)
+            {
+                showerNodeSet.insert(pNode);
+                ++nAboveThresholdShowerMatches;
+            }
+            else
+            {
+                trackNodeSet.insert(pNode);
+                ++nAboveThresholdTrackMatches;
+            }
+        }
+
+        for (const LArHierarchyHelper::MCMatches &mcMatch : matchInfo.GetSubThresholdMatches())
+        {
+            const LArHierarchyHelper::MCHierarchy::Node *pNode{mcMatch.GetMC()};
+            if (pNode->IsLeadingLepton())
+                hasLeadingLepton = 1;
+            const int pdg{std::abs(pNode->GetParticleId())};
+            if (pdg == PHOTON || pdg == E_MINUS)
+                showerNodeSet.insert(pNode);
+            else
+                trackNodeSet.insert(pNode);
+        }
+
+        for (const LArHierarchyHelper::MCMatches &mcMatch : matchInfo.GetUnmatchedMC())
+        {
+            const LArHierarchyHelper::MCHierarchy::Node *pNode{mcMatch.GetMC()};
+            if (pNode->IsLeadingLepton())
+                hasLeadingLepton = 1;
+            const int pdg{std::abs(pNode->GetParticleId())};
+            if (pdg == PHOTON || pdg == E_MINUS)
+                showerNodeSet.insert(pNode);
+            else
+                trackNodeSet.insert(pNode);
+        }
+
+        const int nTrackNodes{static_cast<int>(trackNodeSet.size())}, nShowerNodes{static_cast<int>(showerNodeSet.size())};
```

Comment:
Why are these ints and not unsigned ints? It looks like their default value would be 0 since they are set from the size of a vector.

Reply:
If I recall correctly, because SetTreeVariable doesn't support unsigned ints